src/Stratis.Bitcoin.Features.RPC/Controllers/FullNodeController.cs

```diff
@@ -185,33 +185,33 @@ public BlockHeaderModel GetBlockHeader(string hash, bool isJsonFormat = true)
         /// <returns>JObject containing a boolean indicating address validity</returns>
         [ActionName("validateaddress")]
         [ActionDescription("Returns information about a bech32 or base58 bitcoin address")]
-        public JObject ValidateAddress(string address)
+        public ValidatedAddress ValidateAddress(string address)
```

Comment:
@DanGould this is already much better. But lets just return JsonResult here. Then, in the method return the boolean like so:

```
var isAddressValid = false;
return JsonResult(isAddressValid);
```

Reply:
There is more to it than just a boolean.
Look here:
https://bitcoin.org/en/developer-reference#validateaddress.
For now that's enough, but I think we can easily add more to it.

Reply:
@bokobza my apologies, I wasn't aware of the bitcoin developer reference! In that case yes returning ValidateAddress is good 👍